### PR TITLE
Find command at install time + trial test for command at test time

### PR DESCRIPTION
**Pull request by plicease**
By default AB doesn't add commands to the PATH for a share install.  It is buried in the `Alien::Base` man page but there is a recipe there:

https://metacpan.org/pod/Alien::Base#bin_dir

The philosophy is not to install tools or libraries in system locations, or to make them available unless you ask for them.  This requires more effort unfortunately for the XS or FFI module using the Alien, but it plays nicer with system integrators.

### CODE REVIEW

File: alienfile [hunk 1]
```diff
@@ -4,6 +4,11 @@ plugin 'PkgConfig' => (
   pkg_name => 'libtiff-4',
 );
 
+plugin 'Probe::CommandLine' => (
+  command   => 'tiffcp',
```
Comment (plicease) [inline alienfile:8]: System integrators like Debian often split the commands from the library packages.  On my system for example libtif.a etc was installs but none of the tools.  So if you really want the tools to be available you need t ocheck for them too.  This is a simple recipe just to check for the existence of the tool, but there are other checks you can do see 
https://metacpan.org/pod/Alien::Build::Plugin::Probe::CommandLine

File: t/1.t [hunk 2]
```diff
@@ -1,4 +1,14 @@
-use IPC::Cmd qw(can_run run);
-use Test::More tests => 1;
+use strict;
+use warnings;
+use Test::More tests => 3;
+use Test::Alien;
```
Comment (plicease) [inline t/1.t:4]: Using Test::Alien to test aliens is highly recommended.  It handles a lot of corner cases (like putting the appropriate bin directories in the PATH for a share install).

File: t/1.t [hunk 3]
```diff
@@ -1,4 +1,14 @@
...
+use Alien::libtiff;
+use File::Temp;
+
+alien_ok 'Alien::libtiff';
+
+my $dest = File::Temp->new( TEMPLATE => 'fooXXXX', SUFFIX => '.tiff' );
+
+run_ok(['tiffcp', 't/x.tiff', "$dest"])
+  ->success;
```
Comment (plicease) [inline t/1.t:13]: run_ok actually runs the tool.  This is good because it makes sure that any dynamic linking is done correctly.

Often you can away with just 'tool --version' or something, but I didn't see an obvious way to do it with this tool so I created a 1x1 pixel tiff image and ran it on that.

**Comment by plicease** [comment]
If your XS or FFI that is consuming this Alien doesn't need the command line tools, you may also want to consider not checking for them, and not testing for them and instead testing using the XS or FFI checks in `Test::Alien`.  This would allow your Alien to install on a system with the dev libraries but not the tools like mine.  Of course if you want and need the tools as well, what I have here should work.

If there is an XS consumer it is highly recommended that you write a test to make sure that it works.  There are a lot of corner cases with dynamic libraries, and it is good to catch any failures when installing the Alien rather than the XS consumer.  I usually write a "toy" XS and run it with `Test::Alien`.  The `Test::Alien` has some examples, but here is a real world example:

https://metacpan.org/source/PLICEASE/Acme-Alien-DontPanic-2.2900/t%2Facme_alien_dontpanic.t

**Comment by carygravel** [comment]
Thanks for your help. That got the ubuntu CI runner working. The windows runner dies now checking for libtiff:

`
Command not found tiffcp at C:/Strawberry/perl/vendor/lib/Alien/Build/Plugin/Probe/CommandLine.pm line 44.
`

I would expect this to trigger downloading and building libtiff. What is preventing that?
